nixos/lib/test-driver/src/test_driver/logger.py

```diff
@@ -48,6 +48,10 @@ def error(self, *args, **kwargs) -> None:  # type: ignore
     def log_test_error(self, *args, **kwargs) -> None:  # type:ignore
         pass
 
+    @abstractmethod
+    def dump(self, *args, **kwargs) -> None:  # type:ignore
```

Comment:
I'm not convinced by this new `dump` function, the `print` function is basically universal to print something out and I don't see the point of having it not have the 'provenance' prefix by default 🤔

I believe the _default_ behavior should be to have the prefix, and maybe expose the non-prefix print as `raw_print` or `direct_print` ?

Reply:
So, on one hand I agree. OTOH I think it's kinda weird to override `print()` and give it special behavior (unless `file=` is set).

For me, this was the less intrusive/controversial change, but I'm very open to do what @bew suggested. What do the maintainers think?